sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1196,6 +1273,11 @@ def task_worker(self):
                             self.port_dict[lport]['forced_tx_disabled'] = False
                             self.log_notice("{}: Tx laser is successfully turned OFF".format(lport))
 
+                        # Skip rest of the pre-init when decommission is required
+                        if not self.is_decomm_pending_for_lport(lport):
```

Comment:
Looks like, when this condition is true (i.e. not in decommission), the below part of CMIS_STATE_DP_PRE_INIT_CHECK will get skipped:

https://github.com/sonic-net/sonic-platform-daemons/blob/8f09f2e50d8ce30644def7dee4a103ed61c112ef/sonic-xcvrd/xcvrd/xcvrd.py#L1281-L1317